source/lib/src/gpu/tabulate.cu

```diff
@@ -4,9 +4,30 @@
 #define MM 4
 #define KK 4
 #define TPB 256
+#if GOOGLE_CUDA
 #define WARP_SIZE 32
+#elif TENSORFLOW_USE_ROCM
+// See https://github.com/pytorch/pytorch/pull/64302
+#define WARP_SIZE warpSize  // = 64 or 32 (Defined in hip_runtime.h)
+#else
+#error "should not touch here"
+#endif
 #define FULL_MASK 0xffffffff
 
+// Copyright 2017 The TensorFlow Authors.
+// Licensed under the Apache License, Version 2.0
+template <typename T>
+__device__ T
+GpuShuffleSync(unsigned mask, T value, int src_lane, int width = warpSize) {
+#if GOOGLE_CUDA
+  return __shfl_sync(mask, value, src_lane, width);
```

Comment:
It seems `__shfl_sync` and `__syncwarp` are new in CUDA 9. Should we update the documentation here from `7` to `9`? https://github.com/deepmodeling/deepmd-kit/blob/339ce4743194632ad6c4a906fb7c9fa4fe41dc32/doc/install/install-from-source.md?plain=1#L77

Reply:
Yes, I think we should update the doc